Buildings/Templates/ChilledWaterPlant/Components/PrimaryPumpGroup/BaseClasses/DedicatedPrimaryPumps.mo

```diff
@@ -0,0 +1,128 @@
+within Buildings.Templates.ChilledWaterPlant.Components.PrimaryPumpGroup.BaseClasses;
+model DedicatedPrimaryPumps
+  replaceable package Medium =
+    Modelica.Media.Interfaces.PartialMedium "Medium in the component";
+
+  // Pump parameters
+  parameter Integer nPum "Number of pumps";
+  parameter Modelica.SIunits.MassFlowRate m_flow_nominal "Nominal mass flow rate per pump";
```

Comment:
The component should allow for pumps with different nominal mass flow rates and flow curves.